File: go/diffie-hellman/diffie_hellman.go
package diffiehellman

import (
	"math/big"
	"math/rand"
	"time"
)

// PrivateKey generates a private key
func PrivateKey(p *big.Int) *big.Int {
	src := rand.NewSource(time.Now().Unix())
	r := rand.New(src)
	return p.Rand(r, p)
}

// PublicKey generates public key
func PublicKey(a, p *big.Int, g int64) *big.Int {

}

// SecretKey returns secret key
func SecretKey(a, B, p *big.Int) *big.Int {
	// return math.Pow(B, a) % p
	return nil
}

// NewPair returns a new pair
func NewPair(p *big.Int, g int64) {
	return
}

File: python/diffie-hellman/diffie_hellman.py
import secrets

def private_key(p):
    if secrets.randbelow(p) > p or p < 1:
        raise Exception("private key must be less than p")
    else:
        return secrets.randbelow(p)


def public_key(p, g, private):
    return (g**private) % p


def secret(p, public, private):
    return (public**private) % p
